### PR TITLE
gcc7 compilation warning(s) fix(es) in L1Trigger/RegionalCaloTrigger

### PR DESCRIPTION
**Pull request by mrodozov**
Fix for compilation warning(s) when compiling with gcc7 and more flags listed here:
https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc6_amd64_gcc700/CMSSW_9_3_X_2017-07-24-2300/L1Trigger/RegionalCaloTrigger

### CODE REVIEW

**Comment by davidlange6** [comment]
please test

**Comment by smuzaffar** [comment]
@rekovic, @mulhearn can you please review and sign this PR?

File: L1Trigger/RegionalCaloTrigger/plugins/MaskedRctInputDigiProducer.cc [hunk 1]
```diff
@@ -272,13 +272,13 @@ MaskedRctInputDigiProducer::produce(edm::Event& iEvent, const edm::EventSetup& i
 	     {
 	       //std::cout << "eta-: mask is " << ecalMask.at(0).at(iphi-1).at(absIeta-1) << std::endl;
 	       energy = ecalMask.at(0).at(iphi-1).at(absIeta-1) * ecalColl[i].sample(nSample).compressedEt();
-	       fineGrain = ecalMask.at(0).at(iphi-1).at(absIeta-1) * ecalColl[i].sample(nSample).fineGrain();
+	       fineGrain = ( ecalMask.at(0).at(iphi-1).at(absIeta-1) * ecalColl[i].sample(nSample).fineGrain() ) != 0;
```
Comment (davidlange6) [inline L1Trigger/RegionalCaloTrigger/plugins/MaskedRctInputDigiProducer.cc]: hi @mrodozov - i think the right fix is 

fineGrain = ( ecalMask.at(0).at(iphi-1).at(absIeta-1) !=0 ) &&  ecalColl[i].sample(nSample).fineGrain() 

[fineGrain() returns a bool]

**Comment by mrodozov** [comment]
updated it as proposed

**Comment by davidlange6** [comment]
please test

**Pull request merged**